blueprints/nvidia-gpu-workload/README.md

```diff
@@ -0,0 +1,220 @@
+# Karpenter Blueprint: Deploy an NVIDIA GPU workload
+
+## Purpose
+
+Karpenter streamlines node lifecycle management, and it can help provide the right compute just-in-time based on your workloads scheduling constraints. This is particularly helpful for your machine learning workflows with variable and heterogeneous compute demands (e.g., NVIDIA GPU-based inference followed by CPU-based plotting). When your Kubernetes workload requires accelerated instance, Karpenter automatically selects the appropriate [Amazon EKS optimized accelerated AMI](https://docs.aws.amazon.com/eks/latest/userguide/eks-optimized-ami.html). 
+
+Therefore, the purpose of this Karpenter blueprint is to demonstrate how to launch a GPU-based workload on Amazon EKS with Karpenter and AL2023 EKS optimized accelerated AMI. This example assumes a simple one-to-one mapping between a Kubernetes Pod and a GPU. This blueprint does not go into the details about GPU sharing techniques such as MiG, time slicing or other software based GPU fractional scheduling.
+
+Before you start seeing Karpenter in action, when using AL2023 you need to deploy a Kubernetes device plugin to advertise GPU information from the host.
+
+## Requirements
+* A Kubernetes cluster with Karpenter installed. You can use the blueprint we've used to test this pattern at the `cluster` folder in the root of this repository.
+
+## Deploy NVIDIA device plugin for Kubernetes
+
+The [NVIDIA device plugin for Kubernetes](https://github.com/NVIDIA/k8s-device-plugin) is used to advertise the number of GPUs on the host to Kubernetes so that this information can be used for scheduling purposes. You can install the NVIDIA device plugin with helm.
+
+To install the device plugin run the following:
+
+```
+helm repo add nvdp https://nvidia.github.io/k8s-device-plugin
+helm repo update
+
+cat << EOF > values.yaml
+gfd:
+  enabled: true
+nfd:
+  enabled: true
+  worker:
+    tolerations:
+      - key: nvidia.com/gpu
+        operator: Exists
+        effect: NoSchedule
+      - operator: "Exists"
+EOF
+
+helm upgrade -i nvdp nvdp/nvidia-device-plugin \
+  --namespace nvidia-device-plugin \
+  --create-namespace \
+  --version 0.17.1 -f values.yaml
+```
+
+Now that you have the device set-up, let’s enable Karpenter to launch NVIDIA GPU instances.
+
+## Create a NodeClass and NodePool with GPU-instances (AL2023)
+
+The following NodeClass, specify the Security Group and Subnet selector, along with AMI. We are using AL2023 here, and when launching an accelerated instance Karpenter will pick the respective EKS optimized accelerated AMI. AL2023 comes packaged with the NVIDIA GPU drivers, and the container runtime is configured out of the box.
+
+The EC2NodeClass we’ll deploy looks like this:
+```
+apiVersion: karpenter.k8s.aws/v1
```

Comment:
Now that I think about this, we'll need to keep this and the file in sync, why don't we create the files in flight? that way we won't have to make sure we keep the manifests in sync.

Reply:
and this apply for all the three files